src/main/java/com/nylas/NylasClient.java

```diff
@@ -183,6 +183,9 @@ ResponseBody download(String authUser, HttpUrl.Builder url) throws IOException,
 
 	Request buildRequest(String authUser, HttpUrl.Builder url, HttpMethod method, RequestBody body, AuthMethod authMethod) {
 		Request.Builder builder = new Request.Builder().url(url.build());
+		if(authMethod == null) {
+			authMethod = AuthMethod.BASIC;
+		}
 		if (authUser != null) {
```

Comment:
maybe check the RequestBody for null if the method isn't a GET

Reply:
This method is package-private access so no one will be using this other than us, so I don't think null checking here is necessary. Also Request.Builder.method() on line 192 has a null check for the body if the method requires a body, so that check is built in.